### ISSUE TITLE
Google snapshot error

### ISSUE THREAD

**Issue by psheridan050** [sabeechen/hassio-google-drive-backup#447]
Copy the info from the p
###### Description:
```
If you have anything else that could help explain what happened, click "Markdown" above and write it here.
```

 Addon version: `0.104.3`
 Home Assistant Version: `2021.8.8`
 Supervisor Version: `2021.06.8`
 Supervisor Channel: `stable`
 Hassos Version: `6.2`
 Docker Version: `20.10.6`
 Architecture: `amd64`
 Machine: `qemux86-64`
 Date: `2021-08-21T15:57:13.954912+00:00`
 Timezone: `America/Denver`
 Failure Time: `2021-08-21T15:55:41.357609+00:00`
 Last Good Sync: `2021-08-21T15:55:40.667920+00:00`
 ###### Exception:
 ```
 
  addon/backup/model/coordinator.py:176 (_sync)
  addon/backup/model/model.py:178 (sync)
  addon/backup/model/model.py:294 (_handleSnapshotDetails)
  addon/backup/model/backupscheme.py:194 (handleNaming)
  addon/backup/model/backupscheme.py:117 (_buildPartitions)
IndexError: list index out of range

 ```
 Snapshots:
 ```
 {
    "HomeAssistant": {
        "snapshots": 0,
        "retained": 0,
        "deletable": 0,
        "name": "HomeAssistant",
        "title": "Home Assistant",
        "latest": null,
        "max": 4,
        "enabled": true,
        "icon": "home-assistant",
        "ignored": 0,
        "size": "0.0 B",
        "ignored_size": "0.0 B",
        "free_space": "275.9 GB"
    },
    "GoogleDrive": {
        "snapshots": 0,
        "retained": 0,
        "deletable": 0,
        "name": "GoogleDrive",
        "title": "Google Drive",
        "latest": null,
        "max": 20,
        "enabled": true,
        "icon": "google-drive",
        "ignored": 0,
        "size": "0.0 B",
        "ignored_size": "0.0 B",
        "free_space": "1.8 TB"
    }
}
 ```
 ###### Config:
 ```
 {
    "Setting.MAX_SNAPSHOTS_IN_GOOGLE_DRIVE": 20,
    "Setting.DAYS_BETWEEN_SNAPSHOTS": 1.0,
    "Setting.IGNORE_OTHER_SNAPSHOTS": true,
    "Setting.IGNORE_UPGRADE_SNAPSHOTS": true,
    "Setting.SNAPSHOT_TIME_OF_DAY": "05:00",
    "Setting.SNAPSHOT_PASSWORD": "REDACTED",
    "Setting.GENERATIONAL_DAYS": 4,
    "Setting.GENERATIONAL_WEEKS": 1,
    "Setting.GENERATIONAL_MONTHS": 1,
    "Setting.GENERATIONAL_DAY_OF_YEAR": 30,
    "Setting.IGNORE_IPV6_ADDRESSES": true,
    "Setting.ALTERNATE_DNS_SERVERS": "172.25.100.100,172.25.100.101"
}
 ```
 ###### Addon Logs:
 ```
 08-21 06:55:38 DEBUG [backup.drive.driverequests] Making Google Drive request: https://www.googleapis.com/drive/v3/files/?q=%271sOj2N8LT-h2HMXi4nI45t3Xx7aRIvsJ5%27+in+parents&fields=nextPageToken%2Cfiles%28id%2Cname%2CappProperties%2Csize%2Ctrashed%2CmimeType%2CmodifiedTime%2Ccapabilities%2Cparents%2CdriveId%29&pageSize=100&supportsAllDrives=true&includeItemsFromAllDrives=true&corpora=allDrives
08-21 06:55:38 ERROR [backup.model.coordinator] 
  addon/backup/model/coordinator.py:176 (_sync)
  addon/backup/model/model.py:178 (sync)
  addon/backup/model/model.py:294 (_handleSnapshotDetails)
  addon/backup/model/backupscheme.py:194 (handleNaming)
  addon/backup/model/backupscheme.py:117 (_buildPartitions)
IndexError: list index out of range

08-21 06:55:38 INFO [backup.model.coordinator] I'll try again in an hour
08-21 07:55:38 DEBUG [backup.model.syncer] Sync requested by Coordinator
08-21 07:55:38 INFO [backup.model.coordinator] Syncing Snapshots
08-21 07:55:38 DEBUG [backup.ha.harequests] Making Hassio request: http://hassio/snapshots
08-21 07:55:38 DEBUG [backup.drive.driverequests] Requesting refreshed Google Drive credentials
08-21 07:55:38 DEBUG [backup.drive.driverequests] Making Google Drive request: https://www.googleapis.com/drive/v3/files/1sOj2N8LT-h2HMXi4nI45t3Xx7aRIvsJ5/?fields=id%2Cname%2CappProperties%2Csize%2Ctrashed%2CmimeType%2CmodifiedTime%2Ccapabilities%2Cparents%2CdriveId&supportsAllDrives=true
08-21 07:55:38 DEBUG [backup.drive.driverequests] Making Google Drive request: https://www.googleapis.com/drive/v3/about?fields=storageQuota
08-21 07:55:39 DEBUG [backup.drive.driverequests] Making Google Drive request: https://www.googleapis.com/drive/v3/files/?q=%271sOj2N8LT-h2HMXi4nI45t3Xx7aRIvsJ5%27+in+parents&fields=nextPageToken%2Cfiles%28id%2Cname%2CappProperties%2Csize%2Ctrashed%2CmimeType%2CmodifiedTime%2Ccapabilities%2Cparents%2CdriveId%29&pageSize=100&supportsAllDrives=true&includeItemsFromAllDrives=true&corpora=allDrives
08-21 07:55:39 ERROR [backup.model.coordinator] 
  addon/backup/model/coordinator.py:176 (_sync)
  addon/backup/model/model.py:178 (sync)
  addon/backup/model/model.py:294 (_handleSnapshotDetails)
  addon/backup/model/backupscheme.py:194 (handleNaming)
  addon/backup/model/backupscheme.py:117 (_buildPartitions)
IndexError: list index out of range

08-21 07:55:39 INFO [backup.model.coordinator] I'll try again in an hour
08-21 08:55:39 DEBUG [backup.model.syncer] Sync requested by Coordinator
08-21 08:55:39 INFO [backup.model.coordinator] Syncing Snapshots
08-21 08:55:39 DEBUG [backup.ha.harequests] Making Hassio request: http://hassio/snapshots
08-21 08:55:39 DEBUG [backup.drive.driverequests] Requesting refreshed Google Drive credentials
08-21 08:55:39 DEBUG [backup.drive.driverequests] Making Google Drive request: https://www.googleapis.com/drive/v3/files/1sOj2N8LT-h2HMXi4nI45t3Xx7aRIvsJ5/?fields=id%2Cname%2CappProperties%2Csize%2Ctrashed%2CmimeType%2CmodifiedTime%2Ccapabilities%2Cparents%2CdriveId&supportsAllDrives=true
08-21 08:55:39 DEBUG [backup.drive.driverequests] Making Google Drive request: https://www.googleapis.com/drive/v3/about?fields=storageQuota
08-21 08:55:40 DEBUG [backup.drive.driverequests] Making Google Drive request: https://www.googleapis.com/drive/v3/files/?q=%271sOj2N8LT-h2HMXi4nI45t3Xx7aRIvsJ5%27+in+parents&fields=nextPageToken%2Cfiles%28id%2Cname%2CappProperties%2Csize%2Ctrashed%2CmimeType%2CmodifiedTime%2Ccapabilities%2Cparents%2CdriveId%29&pageSize=100&supportsAllDrives=true&includeItemsFromAllDrives=true&corpora=allDrives
08-21 08:55:40 ERROR [backup.model.coordinator] 
  addon/backup/model/coordinator.py:176 (_sync)
  addon/backup/model/model.py:178 (sync)
  addon/backup/model/model.py:294 (_handleSnapshotDetails)
  addon/backup/model/backupscheme.py:194 (handleNaming)
  addon/backup/model/backupscheme.py:117 (_buildPartitions)
IndexError: list index out of range

08-21 08:55:40 INFO [backup.model.coordinator] I'll try again in an hour
08-21 09:55:40 DEBUG [backup.model.syncer] Sync requested by Coordinator
08-21 09:55:40 INFO [backup.model.coordinator] Syncing Snapshots
08-21 09:55:40 DEBUG [backup.ha.harequests] Making Hassio request: http://hassio/snapshots
08-21 09:55:40 DEBUG [backup.drive.driverequests] Requesting refreshed Google Drive credentials
08-21 09:55:40 DEBUG [backup.drive.driverequests] Making Google Drive request: https://www.googleapis.com/drive/v3/files/1sOj2N8LT-h2HMXi4nI45t3Xx7aRIvsJ5/?fields=id%2Cname%2CappProperties%2Csize%2Ctrashed%2CmimeType%2CmodifiedTime%2Ccapabilities%2Cparents%2CdriveId&supportsAllDrives=true
08-21 09:55:40 DEBUG [backup.drive.driverequests] Making Google Drive request: https://www.googleapis.com/drive/v3/about?fields=storageQuota
08-21 09:55:41 DEBUG [backup.drive.driverequests] Making Google Drive request: https://www.googleapis.com/drive/v3/files/?q=%271sOj2N8LT-h2HMXi4nI45t3Xx7aRIvsJ5%27+in+parents&fields=nextPageToken%2Cfiles%28id%2Cname%2CappProperties%2Csize%2Ctrashed%2CmimeType%2CmodifiedTime%2Ccapabilities%2Cparents%2CdriveId%29&pageSize=100&supportsAllDrives=true&includeItemsFromAllDrives=true&corpora=allDrives
08-21 09:55:41 ERROR [backup.model.coordinator] 
  addon/backup/model/coordinator.py:176 (_sync)
  addon/backup/model/model.py:178 (sync)
  addon/backup/model/model.py:294 (_handleSnapshotDetails)
  addon/backup/model/backupscheme.py:194 (handleNaming)
  addon/backup/model/backupscheme.py:117 (_buildPartitions)
IndexError: list index out of range

08-21 09:55:41 INFO [backup.model.coordinator] I'll try again in an hour
 ```
 ###### Supervisor Logs:
 ```
 [32m21-08-21 08:28:56 INFO (MainThread) [supervisor.resolution.checks.base] Run check for IssueType.FREE_SPACE/ContextType.SYSTEM[0m
[32m21-08-21 08:28:56 INFO (MainThread) [supervisor.resolution.checks.base] Run check for IssueType.PWNED/ContextType.ADDON[0m
[32m21-08-21 08:28:56 INFO (MainThread) [supervisor.resolution.checks.base] Run check for IssueType.SECURITY/ContextType.CORE[0m
[32m21-08-21 08:28:56 INFO (MainThread) [supervisor.resolution.check] System checks complete[0m
[32m21-08-21 08:28:56 INFO (MainThread) [supervisor.resolution.evaluate] Starting system evaluation with state CoreState.RUNNING[0m
[32m21-08-21 08:28:56 INFO (MainThread) [supervisor.resolution.evaluate] System evaluation complete[0m
[32m21-08-21 08:28:56 INFO (MainThread) [supervisor.resolution.fixup] Starting system autofix at state CoreState.RUNNING[0m
[32m21-08-21 08:28:56 INFO (MainThread) [supervisor.resolution.fixup] System autofix complete[0m
[32m21-08-21 08:30:06 INFO (MainThread) [supervisor.updater] Fetching update data from https://version.home-assistant.io/stable.json[0m
[32m21-08-21 08:30:13 INFO (MainThread) [supervisor.homeassistant.api] Updated Home Assistant API token[0m
[32m21-08-21 08:55:39 INFO (MainThread) [supervisor.api.middleware.security] /snapshots access from cebe7a76_hassio_google_drive_backup[0m
[32m21-08-21 09:00:14 INFO (MainThread) [supervisor.homeassistant.api] Updated Home Assistant API token[0m
[32m21-08-21 09:23:37 INFO (MainThread) [supervisor.host.info] Updating local host information[0m
[32m21-08-21 09:23:37 INFO (MainThread) [supervisor.host.services] Updating service information[0m
[32m21-08-21 09:23:37 INFO (MainThread) [supervisor.host.network] Updating local network information[0m
[32m21-08-21 09:23:39 INFO (MainThread) [supervisor.host.sound] Updating PulseAudio information[0m
[32m21-08-21 09:23:39 INFO (MainThread) [supervisor.host] Host information reload completed[0m
[32m21-08-21 09:28:56 INFO (MainThread) [supervisor.resolution.check] Starting system checks with state CoreState.RUNNING[0m
[32m21-08-21 09:28:56 INFO (MainThread) [supervisor.resolution.checks.base] Run check for IssueType.FREE_SPACE/ContextType.SYSTEM[0m
[32m21-08-21 09:28:56 INFO (MainThread) [supervisor.resolution.checks.base] Run check for IssueType.PWNED/ContextType.ADDON[0m
[32m21-08-21 09:28:56 INFO (MainThread) [supervisor.resolution.checks.base] Run check for IssueType.SECURITY/ContextType.CORE[0m
[32m21-08-21 09:28:56 INFO (MainThread) [supervisor.resolution.check] System checks complete[0m
[32m21-08-21 09:28:56 INFO (MainThread) [supervisor.resolution.evaluate] Starting system evaluation with state CoreState.RUNNING[0m
[32m21-08-21 09:28:56 INFO (MainThread) [supervisor.resolution.evaluate] System evaluation complete[0m
[32m21-08-21 09:28:56 INFO (MainThread) [supervisor.resolution.fixup] Starting system autofix at state CoreState.RUNNING[0m
[32m21-08-21 09:28:56 INFO (MainThread) [supervisor.resolution.fixup] System autofix complete[0m
[32m21-08-21 09:30:14 INFO (MainThread) [supervisor.homeassistant.api] Updated Home Assistant API token[0m
[32m21-08-21 09:55:40 INFO (MainThread) [supervisor.api.middleware.security] /snapshots access from cebe7a76_hassio_google_drive_backup[0m
[32m21-08-21 09:57:13 INFO (MainThread) [supervisor.api.middleware.security] /supervisor/logs access from cebe7a76_hassio_google_drive_backup[0m

 ```
 ###### Home Assistant Core Logs:
 ```
 [33m2021-08-21 15:55:20 WARNING (MainThread) [homeassistant.components.sensor] Updating opensky sensor took longer than the scheduled update interval 0:00:12[0m
[31m2021-08-21 15:55:20 ERROR (MainThread) [homeassistant.helpers.entity] Update for sensor.opensky fails
Traceback (most recent call last):
  File "/usr/src/homeassistant/homeassistant/helpers/entity.py", line 446, in async_update_ha_state
    await self.async_device_update()
  File "/usr/src/homeassistant/homeassistant/helpers/entity.py", line 662, in async_device_update
    await task
  File "/usr/local/lib/python3.9/concurrent/futures/thread.py", line 52, in run
    result = self.fn(*self.args, **self.kwargs)
  File "/usr/src/homeassistant/homeassistant/components/opensky/sensor.py", line 140, in update
    states = self._session.get(OPENSKY_API_URL).json().get(ATTR_STATES)
  File "/usr/local/lib/python3.9/site-packages/requests/models.py", line 900, in json
    return complexjson.loads(self.text, **kwargs)
  File "/usr/local/lib/python3.9/site-packages/simplejson/__init__.py", line 525, in loads
    return _default_decoder.decode(s)
  File "/usr/local/lib/python3.9/site-packages/simplejson/decoder.py", line 370, in decode
    obj, end = self.raw_decode(s)
  File "/usr/local/lib/python3.9/site-packages/simplejson/decoder.py", line 400, in raw_decode
    return self.scan_once(s, idx=_w(s, idx).end())
simplejson.errors.JSONDecodeError: Expecting value: line 1 column 1 (char 0)[0m
[33m2021-08-21 15:55:42 WARNING (MainThread) [homeassistant.helpers.entity] Update of sensor.opensky is taking over 10 seconds[0m
[33m2021-08-21 15:55:44 WARNING (MainThread) [homeassistant.components.sensor] Updating opensky sensor took longer than the scheduled update interval 0:00:12[0m
[33m2021-08-21 15:55:56 WARNING (MainThread) [homeassistant.components.sensor] Updating opensky sensor took longer than the scheduled update interval 0:00:12[0m
[33m2021-08-21 15:56:08 WARNING (MainThread) [homeassistant.components.sensor] Updating opensky sensor took longer than the scheduled update interval 0:00:12[0m
[33m2021-08-21 15:56:20 WARNING (MainThread) [homeassistant.components.sensor] Updating opensky sensor took longer than the scheduled update interval 0:00:12[0m
[33m2021-08-21 15:56:32 WARNING (MainThread) [homeassistant.components.sensor] Updating opensky sensor took longer than the scheduled update interval 0:00:12[0m
[33m2021-08-21 15:56:44 WARNING (MainThread) [homeassistant.components.sensor] Updating opensky sensor took longer than the scheduled update interval 0:00:12[0m
[33m2021-08-21 15:56:56 WARNING (MainThread) [homeassistant.components.sensor] Updating opensky sensor took longer than the scheduled update interval 0:00:12[0m
[33m2021-08-21 15:57:08 WARNING (MainThread) [homeassistant.components.sensor] Updating opensky sensor took longer than the scheduled update interval 0:00:12[0m

**Comment by sabeechen** [comment]
You found a bug!  I haven't had a good bug in a while.  It looks like this will happen if you have generational snapshots enabled and don't have any snapshots in Home Asisstant or Google Drive yet.

Until I get a fix out, I think you could work around this by doing the following:
- Modify your settings to:
  - Turn off generational snapshots
  - Set ```Snapshots in Google Drive``` and ```Snapshots in Home Assistant``` to something high, like 1000, so it doesn't delete any of your existing snapshots.
 - Save the settings, and let the addon upload a snapshot to Google Drive, creating a snapshot if necessary.
- Return your settings to their original configuration:
  - ```Generational Days```: 4
  - ```Generational Weeks```: 1
  - ```Generational Months```: 1
  - ```Snapshots in Google Drive```: 20
  - ```Snapshots in Home Assistant```: 4

**Comment by psheridan050** [comment]
Looks like that did the trick.  Thank you for your help.

**Comment by sabeechen** [comment]
I'll leave this open until I release the fix, just in case other run into it and come looking to see if it was reported.

**Comment by sabeechen** [comment]
It took a bit for me to get through everything in the latest release, but I've fixed this in the just released v0.105.1.
You can wait for HA to notice the new version, or go to "Supervisor" > "Addon Store" > "..." > "Reload"  to force a check.
Please feel free to re-open this issue if you think there is something else I can help with.
